### PR TITLE
StackSpace: use std::aligned_alloc()

### PR DESCRIPTION
**Pull request by chfast**
Use `std::aligned_alloc()` (and `_aligned_malloc()` on MSVC) to allocate the memory for EVM stack.

This removes unnecessary initialization of the memory to zero (closes https://github.com/ethereum/evmone/issues/403).

The memory alignment is set to 32 bytes (as previously) but allows transitioning to "over-aligned" allocation for additional optimization
(see https://github.com/ethereum/evmone/issues/781).

### CODE REVIEW

File: lib/evmone/execution_state.hpp [hunk 1]
```diff
@@ -27,20 +28,47 @@ using intx::uint256;
...
+            _aligned_free(p);
+#else
+            std::free(p);
+#endif
+        }
+    };
+
+    /// The storage allocated for maximum possible number of items.
+    /// Items are aligned to 256 bits for better packing in cache lines.
+    std::unique_ptr<uint256, Deleter> m_stack_space{allocate()};
```
Comment (battlmonstr) [inline lib/evmone/execution_state.hpp:60]: I'd put the `allocate()` call in a constructor.

File: lib/evmone/execution_state.hpp [hunk 2]
```diff
@@ -27,20 +28,47 @@ using intx::uint256;
...
 public:
     /// The maximum number of EVM stack items.
     static constexpr auto limit = 1024;
 
+
     /// Returns the pointer to the "bottom", i.e. below the stack space.
     [[nodiscard, clang::no_sanitize("bounds")]] uint256* bottom() noexcept
     {
-        return m_stack_space - 1;
+        return m_stack_space.get() - 1;
```
Comment (battlmonstr) [inline lib/evmone/execution_state.hpp:70]: Why does it need to go outside the memory bounds?

**Review by battlmonstr — APPROVED** [review]

File: lib/evmone/execution_state.hpp (hunk 1 above)
Comment (chfast) [reply lib/evmone/execution_state.hpp]: Added constructor.

Comment (battlmonstr) [reply lib/evmone/execution_state.hpp]: :+1:

File: lib/evmone/execution_state.hpp [hunk 3]
```diff
@@ -27,20 +28,48 @@ using intx::uint256;
...
+
+    /// The storage allocated for maximum possible number of items.
+    /// Items are aligned to 256 bits for better packing in cache lines.
+    std::unique_ptr<uint256, Deleter> m_stack_space;
+
 public:
     /// The maximum number of EVM stack items.
     static constexpr auto limit = 1024;
 
+    StackSpace() noexcept : m_stack_space{allocate()} {}
```
Comment (gumb0) [inline lib/evmone/execution_state.hpp:66]: I wonder if 
```cpp
std::unique_ptr<uint256[]>(new (std::align_val_t(sizeof(uint256)) uint256[limit]);
``` 
achieves the same.

Comment (chfast) [reply lib/evmone/execution_state.hpp:66]: It looks like, however this specifies the array element alignment, so I wonder what happens if you specify 64 for uint256. I plan to align the allocation to `2048*32`.

**Pull request merged**